src/main/java/de/tum/in/www1/artemis/repository/ParticipationRepository.java

```diff
@@ -53,14 +53,14 @@
     @Query("select distinct participation from Participation participation left join fetch participation.results par where participation.exercise.id = :#{#exerciseId} and participation.student.id = :#{#studentId} and (par.id = (select max(id) from participation.results) or par.id = null)")
     Optional<Participation> findByExerciseIdAndStudentIdWithLatestResult(@Param("exerciseId") Long exerciseId, @Param("studentId") Long studentId);
 
-    @Query("select distinct participation from Participation participation left join fetch participation.submissions submission left join fetch submission.result result where participation.exercise.id = :#{#exerciseId} and submission.submitted = true and result is null")
-    List<Participation> findByExerciseIdWithEagerSubmittedSubmissionsWithoutResults(@Param("exerciseId") Long exerciseId);
+    @Query("select distinct participation from Participation participation left join fetch participation.submissions submission left join fetch submission.result result where participation.exercise.id = :#{#exerciseId} and submission.submitted = true and (result is null or result.assessmentType = 'AUTOMATIC')")
+    List<Participation> findByExerciseIdWithEagerSubmittedSubmissionsWithoutManualResults(@Param("exerciseId") Long exerciseId);
 
     @Query("select distinct participation from Participation participation left join fetch participation.results where participation.id = :#{#participationId}")
     Optional<Participation> findByIdWithEagerResults(@Param("participationId") Long participationId);
 
-    @Query("select distinct participation from Participation participation left join fetch participation.submissions where participation.id = :#{#participationId}")
-    Optional<Participation> findByIdWithEagerSubmissions(@Param("participationId") Long participationId);
+    @Query("select distinct participation from Participation participation left join fetch participation.submissions s left join fetch s.result left join fetch participation.results where participation.id = :#{#participationId}")
```

Comment:
Can we use the `EntityGraph` annotation for this?
I think this whole request could be converted to a jpa request without an explict `Query`.

```
@EntityGraph(attributePaths = {"submissions", "results})
Optional<Participation> findByIdWithEagerSubmissionsAndResults(Long id);
```

Reply:
You are right. I changed it to
```
@EntityGraph(attributePaths = { "submissions", "submissions.result", "results" })
Optional<Participation> findWithEagerSubmissionsAndResultsById(Long participationId);
```
Unfortunately, the method name has to end with `byId` to make it work.